R/dependencies.R

```diff
@@ -157,6 +157,28 @@ yamlDeps <- function(yaml) {
   )
 }
 
+stripAltEngines <- function(file, encoding) {
+  contents <- readLines(file, encoding = encoding)
+
+  # generate a list of all the headers
+  engineHeaders <- which(grepl("^## --.*engine=", contents))
+  allHeaders <- c(which(grepl("^## --", contents)), length(contents))
+
+  # calculate the end of each alternate engine code block (the beginning of the
+  # very next code block)
+  engineEnds <- vapply(engineHeaders, function(x) {
+    allHeaders[min(which(allHeaders > x))] - 1
+  }, 0)
+
+  # exclude the alternate engine code block lines
+  regions <- rep.int(TRUE, length(contents))
+  for (h in seq_along(engineHeaders)) {
+    regions[engineHeaders[[h]]:engineEnds[[h]]] <- FALSE
+  }
+
+  writeLines(contents[regions], file)
```

Comment:
This won't happen because `regions` is initialized to be the same length as `contents` here:

https://github.com/rstudio/packrat/pull/344/commits/2112fd28186ecc28f40e7334ba0de005fc9f3f9e#diff-a13fefafa2442d59c624b76872450cfcR174

and we never change its length. To be safe, though, I've added tests to ensure we behave sanely in the scenarios in which we have no R chunks, and when we have only R chunks.

https://github.com/rstudio/packrat/pull/344/commits/3e6aa356d02ea3759c70d0f3b5686c8ad23b067a